ddtrace/tracer.py

```diff
@@ -447,6 +447,8 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
                 _check_pid=False,
             )
 
+            span.set_metric(KEEP_SPANS_RATE_KEY, self.writer.keep_rate)
```

Comment:
I'm curious as to why we submit this value as a span tag as opposed to a normal metric through the metrics pipeline. The rate that we keep isn't particular to any certain kind of span or service so it seems kind of expensive to have a tag on every root span with this data. The way it's written right now, we're going to set the same value on each root span in the payload whereas we could just report the metric once with the metrics pipeline.

Is the goal to make this average per operation_name eventually?

Reply:
The reason we attach the rate to top-level spans is that in the backend it will help us aggregate and display those rates along with other existing rates. This might change in the future.